src/myinfo-types.ts

```diff
@@ -1,160 +1,391 @@
 export enum MyInfoSource {
-    GovtVerified = '1',
-    UserProvided = '2',
-    NotApplicable = '3',
-    SingPassVerified = '4',
+  GovtVerified = '1',
+  UserProvided = '2',
+  NotApplicable = '3',
+  SingPassVerified = '4',
 }
 
-export enum MyInfoSex {
-    Male = 'M',
-    Female = 'F',
-    Unknown = 'U',
+interface IMetadata {
+  lastupdated: string
+  source: MyInfoSource
+  classification: 'C'
+  unavailable?: boolean
 }
```

Comment:
We should construct types such that it rules out impossible data objects.

For example, a person may not own a motor vehicle which could make the data item unavailable:

From the specs:
> Note: When a requested data item is unavailable:
> 
> for data item objects, the data item will have the property unavailable with value true.
> In such cases, no additional properties (other than classification, source, and lastupdated) will be provided for the data item.

```typescript
type Metadata = {
  lastupdated: string
  source: MyInfoSource
  classification: 'C'
}

// `unavailable` property doesn't exist when its available
type Unavailable = Metadata & {
  unavailable: false
}

type AvailableMyInfoVehicle = Metadata & { /* all the properties of a motor vehicle */ }

type IMyInfoVehicle = IUnavailable | AvailableMyInfoVehicle
```

Then the user only need to write the type guard as `if(!data.unavailable) { .. }` for the compiler to infer that the type is `AvailableMyInfoVehicle`

Reply:
On second thought, for this example it might complain that `unavailable` may not exist. I'm of two minds whether we should leave it like this, or conveniently (but incorrectly) add `unavailable: false` to `Metadata` type for convenience.

Reply:
`unavailable: true` seems to be a guarantee that the data _doesn't_ exist, whereas `unavailable: false` or `unavailable: undefined` don't seem to be guarantees that the data _does_ exist. this means when trying to access the value, we'd always have to do something like `if (data.meredekagen?.eligibility?.value)`, regardless of the value of `unavailable`. hence I'm not sure whether typing the `unavailable` property tightly is of much value (although we can always do it for correctness)